### PR TITLE
Upgrade to opentelemetry 0.22

### PR DESCRIPTION
**Pull request by djc**
## Motivation

Would like to upgrade to opentelemetry 0.22.

## Solution

This seems mostly straightforward. Upstream deprecated `opentelemetry_jaeger::new_pipeline()` (which is used in an example) in favor of the OTLP collector for Jaeger; given that we already have an OTLP example should we just delete it?

### CODE REVIEW

**Review by jtescher — APPROVED** [review]
Looks good, seems like the jaeger -> otlp exporter switch needs to happen to avoid the deprecation errors

**Comment by djc** [comment]
I was suggesting just removing the Jaeger example -- are you saying it should be switched to OTLP? (I noticed there's already a fairly elaborate OTLP setup in one of the other examples, so not sure it makes sense to have both.)

**Comment by jtescher** [comment]
Can remove too, whichever is easier

**Comment by rbtcollins** [comment]
I think like asked for in #99 the silent API break of i64 histograms should be added to a Changelog entry. 

Something like
```
Breaking Changes

* Upgrade to v0.22.0 of opentelemetry. For list of breaking changes in OpenTelemetry, see the [v0.22.0 changelog](https://github.com/open-telemetry/opentelemetry-rust/blob/v0.22.0/opentelemetry/CHANGELOG.md).
* ia64 histograms will silently downgrade to k:v exports. This is not statically detected. Code like the following produces ia64 histograms:

\```
        tracing::info!(histogram.abcdefg_auenatsou = -19_i64);
\```
```

File: Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ once_cell = "1.13.0"
 
 # Fix minimal-versions
 async-trait = { version = "0.1.56", optional = true }
+futures-util = { version = "0.3.17", optional = true }
```
[the remark this answers is not in the document]
Comment (jtescher) [reply Cargo.toml:38]: yeah true

**Comment by djc** [comment]
I've added a changelog entry and bumped the version number. @jtescher do you think this is good to release?

**Comment by rbtcollins** [comment]
CI failure is
https://github.com/tokio-rs/tracing-opentelemetry/actions/runs/8055902534/job/22003813184?pr=100#step:5:399-415

Cow imported twice.

**Comment by djc** [comment]
Yeah, was looking at this last night but it's not actually clear to me where the second import is.

**Comment by valkum** [comment]
> Yeah, was looking at this last night but it's not actually clear to me where the second (or rather, the first) import is.

A `borrow::Cow` was recently added by a PR to `layer.rs`. Wondering if this is some rebase GitHub thing. The CI used 8b31f0a1fba35c91dd26972473d56ebb5036b344 which is not the HEAD of your branch.

**Comment by djc** [comment]
> > Yeah, was looking at this last night but it's not actually clear to me where the second (or rather, the first) import is.
> 
> A `borrow::Cow` was recently added by a PR to `layer.rs`. Wondering if this is some rebase GitHub thing. The CI used [8b31f0a](https://github.com/tokio-rs/tracing-opentelemetry/commit/8b31f0a1fba35c91dd26972473d56ebb5036b344) which is not the HEAD of your branch.

Ahh, that explains it, thanks!

**Comment by vriesk** [comment]
Hi! Can this be merged now?

**Comment by djc** [comment]
(@jtescher also, would be happy to help maintaining this if you're running short on time/energy.)

**Comment by jtescher** [comment]
@djc thanks! I don't have admin on this repo, could get added in discord maybe? but I made you an owner on crates.io 👍

**Pull request merged**